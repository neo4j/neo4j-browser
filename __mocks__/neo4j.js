/*
 * Copyright (c) 2002-2018 "Neo4j, Inc"
 * Network Engine for Objects in Lund AB [http://neotechnology.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

function integerFn (val) {
  this.val = val
}
integerFn.prototype.toString = function () {
  return this.val.toString()
}

var out = {
  v1: {
    Int: integerFn,
    isInt: function (val) {
      return val instanceof integerFn
    },
    types: {
      Node: function Node (id, labels, properties) {
        this.identity = id
        this.labels = labels
        this.properties = properties
      },
      Relationship: function Relationship (id, start, end, type, properties) {
        this.identity = id
        this.start = start
        this.end = end
        this.type = type
        this.properties = properties
      },
      Path: function Path (start, end, segments) {
        this.start = start
        this.end = end
        this.segments = segments
        this.length = segments.length
      },
      PathSegment: function PathSegment (start, relationship, end) {
        this.start = start
        this.relationship = relationship
        this.end = end
      }
    },
    Integer: function Integer ({ low, high }) {
      this.low = low
      this.high = high
    }
  }
}

out.v1.types.Node.prototype.toString = function () {
  return 'node'
}
out.v1.types.Relationship.prototype.toString = function () {
  return 'rel'
}
out.v1.types.Path.prototype.toString = function () {
  return 'path'
}
out.v1.types.PathSegment.prototype.toString = function () {
  return 'pathsegment'
}
out.v1.Integer.prototype.toInt = function () {
  return this.low
}
out.v1.int = val => new out.v1.Integer(val)

module.exports = out
