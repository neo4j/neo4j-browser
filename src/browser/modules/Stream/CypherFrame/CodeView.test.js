/*
 * Copyright (c) 2002-2020 "Neo4j,"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import { render } from '@testing-library/react'
import neo4j from 'neo4j-driver'

import {
  CodeViewComponent as CodeView,
  CodeStatusbarComponent as CodeStatusbar
} from './CodeView'

describe('CodeViews', () => {
  describe('CodeView', () => {
    test('displays nothing if not successful query', () => {
      // Given
      const props = { request: { status: 'error' } }

      // When
      const { container } = render(<CodeView {...props} />)

      // Then
      expect(container).toMatchSnapshot()
    })
    test('displays request and response info if successful query', () => {
      // Given
      const props = {
        query: 'MATCH xx0',
        request: {
          status: 'success',
          result: {
            summary: {
              server: {
                version: 'xx1',
                address: 'xx2'
              }
            },
            records: [{ res: 'xx3' }]
          }
        }
      }

      // When
      const { container } = render(<CodeView {...props} />)

      // Then
      expect(container).toMatchSnapshot()
    })
  })
  describe('CodeStatusbar', () => {
    test('displays no statusBarMessage', () => {
      // Given
      const props = { result: {}, maxRows: 0 }

      // When
      const { container } = render(<CodeStatusbar {...props} />)

      // Then
      expect(container).toMatchSnapshot()
    })
    test('displays statusBarMessage', () => {
      // Given
      const props = {
        query: 'MATCH xx0',
        status: 'success',
        result: {
          summary: {
            resultAvailableAfter: neo4j.int(5),
            resultConsumedAfter: neo4j.int(5)
          },
          maxRows: 100,
          records: [{ res: 'xx3' }]
        }
      }

      // When
      const { container } = render(<CodeStatusbar {...props} />)

      // Then
      expect(container).toMatchSnapshot()
    })
  })
})
