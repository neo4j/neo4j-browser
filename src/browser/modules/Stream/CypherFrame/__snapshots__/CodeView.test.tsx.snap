// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeViews CodeStatusbar displays no statusBarMessage 1`] = `
<div>
  <div
    class="sc-ckVGcZ fsAzIa"
  >
    <div
      class="sc-fBuWsC kYaBfc"
    />
  </div>
</div>
`;

exports[`CodeViews CodeStatusbar displays statusBarMessage 1`] = `
<div>
  <div
    class="sc-ckVGcZ fsAzIa"
  >
    <div
      class="sc-fBuWsC kYaBfc"
    >
      Started streaming 1 records after 5 ms and completed after 10  ms.
    </div>
  </div>
</div>
`;

exports[`CodeViews CodeView displays nothing if not successful query 1`] = `<div />`;

exports[`CodeViews CodeView displays request and response info if successful query 1`] = `
<div>
  <div
    class="sc-htpNat fsSTsy"
  >
    <table
      class="sc-gPEVay jOwyJh"
    >
      <tbody
        class="sc-iRbamj tSApQ"
      >
        <tr
          class="sc-gipzik eiuiza"
        >
          <td
            class="sc-csuQGl kaqKNb"
          >
            Server version
          </td>
          <td
            class="sc-Rmtcm gwGgYM"
          >
            xx1
          </td>
        </tr>
        <tr
          class="sc-gipzik eiuiza"
        >
          <td
            class="sc-csuQGl kaqKNb"
          >
            Server address
          </td>
          <td
            class="sc-Rmtcm gwGgYM"
          >
            xx2
          </td>
        </tr>
        <tr
          class="sc-gipzik eiuiza"
        >
          <td
            class="sc-csuQGl kaqKNb"
          >
            Query
          </td>
          <td
            class="sc-Rmtcm gwGgYM"
          >
            MATCH xx0
          </td>
        </tr>
        <tr
          class="sc-gipzik eiuiza"
        >
          <td
            class="sc-csuQGl kaqKNb"
          >
            Summary
            <div
              class="sc-jlyJG xZgSE fa fa-caret-right"
              title="Expand section"
            />
          </td>
          <td
            class="sc-Rmtcm gwGgYM"
          >
            {,  "server": {,    "version": "xx1", ...
          </td>
        </tr>
        <tr
          class="sc-gipzik eiuiza"
        >
          <td
            class="sc-csuQGl kaqKNb"
          >
            Response
            <div
              class="sc-jlyJG xZgSE fa fa-caret-right"
              title="Expand section"
            />
          </td>
          <td
            class="sc-Rmtcm gwGgYM"
          >
            [,  {,    "res": "xx3" ...
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
